openwisp_utils/admin_theme/menu.py

```diff
@@ -0,0 +1,154 @@
+from django.apps import registry
+from django.core.exceptions import ImproperlyConfigured
+from django.urls import reverse
+from django.utils.translation import gettext as _
+
+from ..utils import SortedOrderedDict
+
+MENU = SortedOrderedDict()
+
+
+class BaseMenuItem:
+    def __init__(self, config):
+        if not isinstance(config, dict):
+            raise ImproperlyConfigured(
+                f'"config" should be a type of "dict". Error for config-{config}'
+            )
+
+    def get_context(self, request):
+        return self.create_context(request)
+
+    def create_context(self, request):
+        return {'label': self.label, 'url': self.url, 'icon': self.icon}
+
+
+class MenuItem(BaseMenuItem):
+    def __init__(self, config):
+        super().__init__(config)
+        if config.get('name'):
+            if not isinstance(config['name'], str):
```

Comment:
I would suggest to use always the `.get` method for accessing dict values through keys, even if there existences is checked previously. This would make the code more resilient to changes later in time.
@nemesisdesign @atb00ker what do you think about it?

Reply:
Agreed, I would do something like:

```python
value = config.get('name')
if not value:
  raise # ...
elif not isinstance(value, str):
   raise # ...
# ...
```

@niteshsinha17 you should not call `config.get('name')` and throw away the value like in that `if`, and then access the dict again later. Try to remember to access the dict once and reuse the value.

Reply:
Got it. Thanks :+1: